src/neo-vm/ExecutionEngine.cs

```diff
@@ -1154,6 +1194,84 @@ private bool ExecuteInstruction()
             return true;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private bool ExecuteTry(int catchOffset, int finallyOffset)
+        {
+            if (catchOffset < 0 || finallyOffset < 0) return false;
+            if (finallyOffset + catchOffset <= 0) return false;
+            if (finallyOffset > 0 && catchOffset >= finallyOffset) return false;
+
+            if (CurrentContext.ErrorHandle == null)
+                CurrentContext.ErrorHandle = new ErrorHandle();
+            CurrentContext.ErrorHandle.Push(new TryContext(CurrentContext, catchOffset, finallyOffset));
+            CurrentContext.InstructionPointer += CurrentContext.CurrentInstruction.Size;
+            return true;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal bool ExecuteEndTryCatch(TryState currentState)
+        {
+            var currentTry = CurrentContext.ErrorHandle?.CurContext;
+            if (currentTry is null) return false;
+            if (currentTry.State != currentState) return false;
+
+            if (currentTry.HasFinally)
+            {
+                currentTry.State = TryState.Finally;
+                currentTry.EndTryCatch(CurrentContext.InstructionPointer + CurrentContext.CurrentInstruction.Size);
+                CurrentContext.InstructionPointer = currentTry.FinallyPointer;
+            }
+            else
+            {
+                CurrentContext.ErrorHandle.Pop();
+                int nextOpcodePos = checked(CurrentContext.InstructionPointer + CurrentContext.CurrentInstruction.Size);
+                CurrentContext = InvocationStack.Peek();
+                CurrentContext.InstructionPointer = nextOpcodePos;
+            }
+            return true;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private bool ExecuteEndFinally()
+        {
+            var currentTry = CurrentContext.ErrorHandle?.Pop();
+            if (currentTry is null) return false;
+
+            if (FaultState.Rethrow)
```

Comment:
Could you explain me the use of Rethrow?

Reply:
Case 1:
```js
try
{
    ...
    throw   // When executed here, it'll jmp to finally body. 
}
finally
{
    // After the finally body executed, it will throw the error again unless the error was cathed or vm fault.
}
```
Case 2:
```js
try
{
}
catch
{
  throw  // When executed here, it'll jmp to finally body. 
}
finally
{
   // After the finally body executed, it will throw the error again unless the error was cathed or vm fault.
}
```

Reply:
Thanks!